Stratis.Bitcoin/Features/MemoryPool/MempoolManager.cs

```diff
@@ -9,23 +9,49 @@
 
 namespace Stratis.Bitcoin.Features.MemoryPool
 {
+    /// <summary>
+    /// Memory pool scheduler.
+    /// </summary>
     public class MempoolScheduler : AsyncLock
```

Comment:
I don't mind such a smallish class to be included in the file with something else. I don't mind if you separate such class for whatever reason as well. 

As for interfaces, again I don't mind either. Lot of code has interface in the file with implementation. This makes sense to me if the default implementation is the only implementation. But again, I won't mind separation either.

However, such separation changes are definitely not part of XMLDOC task. So if you want to do this, pls create new tasks.